content/blog/simplify-embedded-projects-with-esp-bsp/index.md

```diff
@@ -0,0 +1,134 @@
+Let’s create a simple project using the `display_audio_photo` example, which is available for the ESP32-S3-BOX-3. This example showcases how to use the display, touch, and audio features.
+
+First, let's see the whole build in Asciinema recording:
+{{< asciinema key="esp-bsp-display-demo" cols="80" rows="24" poster="npt:0:08" >}}
+
+1. **Initialize a New Project**:
+
+   Use the `idf.py` tool to create a new project from the example:
+
+   ```bash
+   idf.py create-project-from-example "espressif/esp-box-3^1.2.0:display_audio_photo"
+   cd display_audio_photo
+   ```
+
+2. **Set the Target**:
+
+   Ensure the correct target is set for your project:
+
+   ```
+   idf.py set-target esp32s3
+   ```
+
+   Note: For users of ESP-IDF 5.3 or newer, it is necessary to add the following dependency with the corrected I2C driver (error message at runtime: `CONFLICT! driver_ng`):
+
+   ```
+   idf.py add-dependency "espressif/esp_codec_dev==1.1.0"
+   ```
+
+3. **Build and Flash the Project**:
+
+   Compile and flash your application to the ESP32-S3-BOX-3:
+
+   ```
+   idf.py build flash monitor
+   ```
+
+   Note: Use `Ctrl+]` to quit the monitor application.
+
+
+### Exploring the Example
+
+Once the application is running, you’ll see the following features in action:
+
+- **Display**: Shows images, text files, and more.
+- **Touch**: Interacts with the display.
+- **Audio**: Plays sound files.
+
+
+### Exploring the Component
+
+The ESP-BSP is included via `main/idf_component.yml` file. Simply by switching to another BSP, you can adapt the project to different hardware. Here’s how you can specify dependencies for different boards:
+
+Example for ESP32-S3-BOX:
+
+```
+## IDF Component Manager Manifest File
+dependencies:
+  espressif/esp-box: "^3.1.0"
+  esp_codec_dev:
+    public: true
+    version: "==1.1.0"
+  ## Required IDF version
+  idf:
+    version: ">=5.0.0"
+```
+
+Example for M5Stack-CoreS3:
+
+```
+## IDF Component Manager Manifest File
+dependencies:
+  espressif/m5stack_core_s3: "^1.1.0"
+  esp_codec_dev:
+    public: true
+    version: "==1.1.0"
+  ## Required IDF version
+  idf:
+    version: ">=5.0.0"
+```
+
+## Conclusion
+
+With ESP-BSP, you can quickly develop and port your applications across various ESP32 boards, saving time and effort. Whether you’re building a new project or upgrading an existing one, ESP-BSP simplifies your development process.
+
+## Useful Links
+
+- [Board Support Packages at Component Registry](https://components.espressif.com/components?q=Board+Support+Package)
+- [ESP-BSP GitHub Repository](https://github.com/espressif/esp-bsp)
+- [ESP-BSP Documentation](https://github.com/espressif/esp-bsp/blob/master/README.md)
+- [ESP-BOX-3 BSP Example](https://components.espressif.com/components/espressif/esp-box-3/versions/1.2.0/examples?language=en)
+- [ESP-IDF Installation Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/index.html)
+- [Wokwi - Online ESP32 Simulator](https://wokwi.com)
```

Comment:
Sorry, I know you like Wokwi a lot, but it seems to be not really related to the article? Of course we can simulate applications based on ESP-BSP in Wokwi, but so can we do for any other applications. I guess we won't be linking Wokwi to every single tutorial we publish?

Reply:
or replace with [Try with ESP Launchpad](https://espressif.github.io/esp-launchpad/?flashConfigURL=https://espressif.github.io/esp-bsp/config.toml&app=display_audio_photo)

Reply:
Launchpad is great idea, and actually BSPs has ready everything even binaries. I've added the button, it works out of box.

Just the issue is that there is Zoom mouse pointer instead of Hand pointer when hover over the image. @f-hollow Do you know how to resolve this issue with mouse pointer?

Reply:
You can use this shortcode:

```html
{{< figure
    src="assets/image.png"
    href="https://www.espressif.com/"
    alt="Image"
    nozoom=true
    caption="Photo by..."
>}}
```

Reply:
@pedrominatel Unfortunately this disable the link, it's not clickable.

Reply:
@igrr I've added a chapter about possibility to simulate the S3-BOX also in context of available IDEs that Espressif or third party provides.

We were discussing with @urish an option how to run the binary directly from the article, the API is not ready yet, in future it might be possible to replace this part directly with simulation of the mentioned example in the article page.

Reply:
Regading Launchpad link, I've tried several options, none of them work.
I've removed the image for the time-being and replaced it by simple link.

Reply:
Me and @georgik had a separate conversation and ended up using the [button shortcode](https://blowfish.page/docs/shortcodes/#button) included in the Blowfish theme.